src/main/java/me/rayzr522/jsonmessage/compat/impl/PlayerConnectImpl17ToFuture.java

```diff
@@ -0,0 +1,45 @@
+package me.rayzr522.jsonmessage.compat.impl;
+
+import me.rayzr522.jsonmessage.ReflectionHelper;
+import me.rayzr522.jsonmessage.compat.PlayerConnectionCompat;
+import org.bukkit.entity.Player;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class PlayerConnectImpl17ToFuture implements PlayerConnectionCompat {
```

Comment:
This is a 1:1 copy of the other one?

Reply:
the classes are different, and the duplication otherwise is intentional -- while it does result in some duplicated code, it's very declarative and avoids too intricate of an abstraction

Reply:
Ah, missed those